pulsar-broker/src/main/java/org/apache/pulsar/common/naming/TopicBundleAssignmentFactory.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.common.naming;
+
+import org.apache.pulsar.broker.PulsarService;
+import org.apache.pulsar.broker.ServiceConfiguration;
+import org.apache.pulsar.common.util.Reflections;
+
+public class TopicBundleAssignmentFactory {
+
+    private static volatile TopicBundleAssignmentStrategy strategy;
+
+    public static TopicBundleAssignmentStrategy create(PulsarService pulsar) {
+        if (strategy != null) {
+            return strategy;
+        }
+        synchronized (TopicBundleAssignmentFactory.class) {
+            if (strategy != null) {
+                return strategy;
+            }
+            ServiceConfiguration conf = pulsar.getConfiguration();
+            try {
+                TopicBundleAssignmentStrategy tempStategy =
+                        Reflections.createInstance(conf.getTopicBundleAssignmentStrategy(),
+                                TopicBundleAssignmentStrategy.class, Thread.currentThread().getContextClassLoader());
+                tempStategy.init(pulsar);
+                strategy = tempStategy;
+                return strategy;
+            } catch (Exception e) {
+                throw new RuntimeException(
+                        "Could not load TopicBundleAssignmentStrategy:" + conf.getTopicBundleAssignmentStrategy(), e);
```

Comment:
This exception could only be caught in `PulsarWebResource#validateNamespaceBundleRange`. So if the strategy class cannot be loaded, each time a REST request comes, an error response will be created.

We should fail fast in this case. So it's better to do the initialization in `PulsarService#start()`.

Reply:
Good catch

Reply:
> This exception could only be caught in `PulsarWebResource#validateNamespaceBundleRange`. So if the strategy class cannot be loaded, each time a REST request comes, an error response will be created.
> 
> We should fail fast in this case. So it's better to do the initialization in `PulsarService#start()`.

This comment seems not resolve yet

Reply:
it's better to do the initialization in `PulsarService#start()`.

Reply:
@xiaotongwang1  Please take a look

Reply:
now i move TopicBundleAssignmentStrategy to   NamespaceBundleFactory 
NamespaceBundleFactory  will init when pusar start

Reply:
It's not straight but just as the explanation above says, the initialization is done when starting the `PulsarService`.